lib/std/crypto/sha2.zig

```diff
@@ -734,11 +740,12 @@ fn roundParam512(a: usize, b: usize, c: usize, d: usize, e: usize, f: usize, g:
     };
 }
 
-/// FIPS 180 Section 5.3.6
-fn sha512iv(digest_len: comptime_int) Iv64 {
-    if (digest_len < 1 or digest_len >= 512 or digest_len == 384) {
-        @compileError("digest_len must be between 1 and 512 but not 384");
-    }
+/// Compute the IV for a truncated version of SHA512 per FIPS 180 Section 5.3.6
```

Comment:
Does that work for the non-truncated version? The comment says it doesn't, the function name suggests it does, the 2nd assert says it does.

Reply:
No, it does not work for the non-truncated versions.

Renamed to `truncatedSha512Iv`.